### PR TITLE
feat: add google provider

### PR DESCRIPTION
**Pull request by lifeizhou-ap**
Added google provider

### CODE REVIEW

File: crates/goose/src/providers/factory.rs [hunk 1]
```diff
@@ -2,6 +2,7 @@ use super::{
     anthropic::AnthropicProvider, base::Provider, configs::ProviderConfig,
     databricks::DatabricksProvider, ollama::OllamaProvider, openai::OpenAiProvider,
 };
+use crate::providers::google::GoogleProvider;
```
Comment (zakiali) [inline crates/goose/src/providers/factory.rs]: move this up into the super for consistency?

**Review by zakiali — APPROVED** [review]
this works for me!

File: download_tokenizer_files.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
     "Xenova/gpt-4o",
     "Xenova/claude-tokenizer",
     "Qwen/Qwen2.5-Coder-32B-Instruct",
+    "Xenova/gemini-nano",
```
Comment (lifeizhou-ap) [inline download_tokenizer_files.py]: Hi @zakiali would you please review this to check whether this file is correct? Thank you

Comment (zakiali) [reply download_tokenizer_files.py]: Doing a bit googling, it looks like gemini 1.5 flash uses the same tokenizer as gemma 2, so we can probably use this instead https://huggingface.co/Xenova/gemma-2-tokenizer

**Pull request merged**